File: 0056-merge-intervals/0056-merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        List<InterVal> interVals=new ArrayList<>();
        for(int[] interval:intervals){
            interVals.add(new InterVal(interval[0],interval[1]));
        }
        Collections.sort(interVals,(a, b)->Integer.compare(a.start,b.start));
        
        List<InterVal> mergedIntervals=new ArrayList<>();
        
        InterVal interval=interVals.get(0);
        int start=interval.start,end=interval.end;
        
        for(int i=1;i<interVals.size();i++){
            interval=interVals.get(i);
            if(interval.start<=end){
                end=Math.max(interval.end,end);
            }else{
                mergedIntervals.add(new InterVal(start,end));
                start=interval.start;
                end=interval.end;
            }
        }
        mergedIntervals.add(new InterVal(start,end));
        
        int[][] result=new int[mergedIntervals.size()][2];
        
        for(int i=0;i<mergedIntervals.size();i++){
            result[i][0]=mergedIntervals.get(i).start;
            result[i][1]=mergedIntervals.get(i).end;
        }
        return result;
    }
    class InterVal{
        int start,end;
        public InterVal(int start, int end) {
            this.start = start;
            this.end = end;
        }
    }
}

File: 0056-merge-intervals/0056-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort(key=lambda x:x[0])
        if len(intervals)<2: return intervals
        ans=[]
        start,end=intervals[0][0],intervals[0][1]
        for i in range(1,len(intervals)):
            if end>=intervals[i][0]:
                end=max(end,intervals[i][1])
            else:
                ans.append([start,end])
                start=intervals[i][0]
                end=intervals[i][1]
            
        ans.append([start,end])
        return ans
        